src/github.com/couchbaselabs/sync_gateway/base/dcp_feed.go

```diff
@@ -0,0 +1,193 @@
+//  Copyright (c) 2014 Couchbase, Inc.
+//  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+//  except in compliance with the License. You may obtain a copy of the License at
+//    http://www.apache.org/licenses/LICENSE-2.0
+//  Unless required by applicable law or agreed to in writing, software distributed under the
+//  License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+//  either express or implied. See the License for the specific language governing permissions
+//  and limitations under the License.
+
+package base
+
+import (
+	"encoding/json"
+	"sync"
+
+	"github.com/couchbase/gomemcached"
+	"github.com/couchbaselabs/go-couchbase/cbdatasource"
+	"github.com/couchbaselabs/walrus"
+)
+
+type couchbaseDCPFeedImpl struct {
```

Comment:
Can this be called simply `couchbaseDCPFeed` rather than sticking `impl` on the end?  This feels non-idiomatic, since the convention in Go is to name interfaces as "-er" or "-able", thus avoiding naming conflicts with structs.